### PR TITLE
Enable Escape & Enter for confirm dialog when delete a component

### PR DESCRIPTION
**Pull request by ColinTree**
Related #1037 

I failed on making the focus onto the button, but i added a key listener to the dialog so that it can detect the hot keys

### CODE REVIEW

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockComponent.java [hunk 1]
```diff
@@ -48,6 +50,7 @@
 import com.google.gwt.user.client.Event;
 import com.google.gwt.user.client.Random;
 import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.Event.NativePreviewEvent;
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockComponent.java]: To keep the ordering of imports consistent, this should come after line 50.

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockComponent.java [hunk 2]
```diff
@@ -248,6 +251,20 @@ public void onClick(ClickEvent event) {
...
       add(contentPanel);
     }
+    @Override
+    protected void onPreviewNativeEvent(NativePreviewEvent event) {
+      super.onPreviewNativeEvent(event);
+      switch (event.getTypeInt()) {
+        case Event.ONKEYDOWN:
+          if (event.getNativeEvent().getKeyCode() == KeyCodes.KEY_ESCAPE) {
+            hide();
+          }else if (event.getNativeEvent().getKeyCode() == KeyCodes.KEY_ENTER) {
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockComponent.java]: Single space between `}else`.

**Comment by ColinTree** [comment]
Done @ewpatton

**Comment by ewpatton** [comment]
@ColinTree Merged. Thanks for your contribution!

**Comment by ColinTree** [comment]
Pleasure 😀

**Pull request merged**